packages/@aws-cdk/aws-lambda/lib/lambda-version.ts

```diff
@@ -67,6 +68,20 @@ export interface VersionOptions extends EventInvokeConfigOptions {
    * @default RemovalPolicy.DESTROY
    */
   readonly removalPolicy?: RemovalPolicy;
+
+  /**
+   * The list of properties under AWS::Lambda::Function that are classified as whether they are version "locked" or not.
+   * A version locked property, will not take effect on previously created Versions. A new Version must be generated for the change to take effect.
+   * On the other hand, properties that are not version locked will take effect on all Versions of the Function.
+   *
+   * All properties need to be classified as version locked or not. Specify this for any properties that are not already classified in the CDK.
+   * NOTE: This only applies when using `Function.currentVersion()`.
+   *
+   * All properties that are part of the UpdateFunctionConfiguration API are locked to the version.
+   * @see https://docs.aws.amazon.com/lambda/latest/dg/API_UpdateFunctionConfiguration.html
+   * @default - use only the version lock classification already encoded in the CDK.
+   */
+  readonly additionalVersionPropsClassification?: { [key: string]: boolean };
```

Comment:
Hmm. Is this not a rather conspicuous place to put this, as well as not being useful if `function.currentVersion()` is being used?

I was thinking this was going to be a static method.

Also, a note in the docstring that "you should only need to call this if you are adding property overrides for properties not in the CDK model" or something would go a long way towards informing people they're free to ignore this 99.9999% of the time.

Reply:
> Is this not a rather conspicuous place to put this, as well as not being useful if function.currentVersion() is being used?

This can also be configured on `currentVersionOptions` property on `FunctionProps`.

> I was thinking this was going to be a static method.

While it's technically correct that this can be configured globally, I'd rather not introduce yet another place for configuration, esp since this is likely never going to be used.